jgrapht-core/src/main/java/org/jgrapht/traverse/LexBreadthFirstIterator.java

```diff
@@ -66,56 +67,71 @@
      * Contains current vertex of the {@code graph}.
      */
     private V current;
-
+    
     /**
-     * Creates new lexicographical breadth-first iterator for {@code graph}.
-     *
-     * @param graph the graph to be iterated.
+     * Lookup tables for a static vertex numbering used to store the vertices.
      */
-    public LexBreadthFirstIterator(Graph<V, E> graph)
-    {
-        super(graph);
-        GraphTests.requireUndirected(graph);
-        bucketList = new BucketList(graph.vertexSet());
-    }
+    private Map<Integer, V> vertexNumbering = new HashMap<Integer, V>();
+    private Map<V, Integer> inverseVertexNumbering = new HashMap<V, Integer>();
+    
+    /**
+     * Lookup tables for the sorted neighborhoods according to the imposed tiebreaking orders.
+     */
+    private Map<V, List<V>> sortedNeighbors = null;
+    private Map<V, List<V>> sortedNeighborsB = null;
     
     /**
      * Creates new lexicographical breadth-first iterator for {@code graph}.
      *
      * @param graph the graph to be iterated.
-     * @param startingVertex the initial vertex.
      */
-    public LexBreadthFirstIterator(Graph<V, E> graph, V startingVertex) {
+    public LexBreadthFirstIterator(Graph<V, E> graph)
+    {
         super(graph);
         GraphTests.requireUndirected(graph);
-
-        Set<V> copyOfSet = new HashSet<>(graph.vertexSet());
-        bucketList = new BucketList(copyOfSet, startingVertex);
+        labelVertices(graph.vertexSet());
+        bucketList = new BucketList(graph.vertexSet());
+        
+        // Precompute sorted neighborhoods
+        this.sortedNeighbors = computeSortedNeighborhoods(graph, vertexNumbering);
     }
     
-    
     /**
      * Creates new lexicographical breadth-first iterator with a static priority list for {@code graph}.
      *
-     * @param graph the graph to be iterated.
-     * @param priority the vertex array sorted by their priorities.
+     * @param graph The graph to be iterated.
+     * @param priority A mapping containing the priorities of the vertices. The priority mapping must be a permutation of $\{0, ..., n - 1\}$ where $n$ is the number of vertices of the graph.
```

Comment:
linebreak

Reply:
Weird, the autoformatter has now changed this. I was sure i ran it before...
(Done via autoformatter)